ServiceTier/Services/LogInService.cs

```diff
@@ -33,22 +40,30 @@ public string CreateToken(LoginDTO model)
 
             if (user == null)
             {
-                throw new UserNotFoundException();
+                throw new LoginFailedException();
             }
 
             if (!userManager.UserManager.CheckPasswordAsync(user, model.Password).Result)
             {
                 throw new LoginFailedException();
             }
 
-            var token = JWTManager.GenerateToken(user);
+            var token = jwtManager.GenerateToken(user);
 
             return token;
         }
 
         public string CreateTokenWithGoogle(SocialAuthDTO accessToken)
         {
-            string userInfoResponse = Client.GetStringAsync($"{Constants.GoogleResponsePath}{accessToken.AccessToken}").Result;
+            string userInfoResponse = null;
+            try
+            {
+                userInfoResponse = Client.GetStringAsync($"{Constants.GoogleResponsePath}{accessToken.AccessToken}").Result;
+            }
+            catch (AggregateException)
```

Comment:
Aggregate exception is usually too *ambiguous* to catch... the whole point of that exception is that it groups different other exceptions that may occur simultaneously. It should have "inner exception" in ti, and that is actually exception you a interested in.
so here you probably should add additional exception filter for that inner exception.

Reply:
ok, will try...

Reply:
There is wrote that inner exception is HTTPRequestException, but I can`t catch this exception, I only AggergateException can catch.
![image](https://user-images.githubusercontent.com/37620894/50975169-945a4e00-14f5-11e9-8dd8-2b41e18edd34.png)
![image](https://user-images.githubusercontent.com/37620894/50975196-a4722d80-14f5-11e9-909a-be67087fdc31.png)